### ISSUE TITLE
测试结果和验证结果是nan

### ISSUE THREAD

**Issue by Little-girl-1992** [macanv/BERT-BiLSTM-CRF-NER#23]
INFO:tensorflow:***** Eval results *****
INFO:tensorflow:  eval_f = nan
INFO:tensorflow:  eval_precision = nan
INFO:tensorflow:  eval_recall = nan
INFO:tensorflow:  global_step = 327
INFO:tensorflow:  loss = 2.0464203

INFO:tensorflow:***** Predict results *****
INFO:tensorflow:  eval_f = nan
INFO:tensorflow:  eval_precision = nan
INFO:tensorflow:  eval_recall = nan
INFO:tensorflow:  global_step = 327
INFO:tensorflow:  loss = 2.4771109

entity level
processed 214543 tokens with 7450 phrases; found: 7740 phrases; correct: 6676.
accuracy:  98.83%; precision:  86.25%; recall:  89.61%; FB1:  87.90
              LOC: precision:  87.66%; recall:  89.81%; FB1:  88.72  3549
              ORG: precision:  77.08%; recall:  85.69%; FB1:  81.15  2408
              PER: precision:  95.85%; recall:  93.90%; FB1:  94.87  1783

**Comment by zuchunshan** [comment]
我也遇到了类似问题，正在排查原因。
我是在多核CPU上运行的，GPU环境不方便升级到CUDA9.0。

**Comment by makotohomula** [comment]
我遇到了类似问题，清除output里面的输出，重新运行时得到了解决

**Comment by newwaylw** [comment]
我也遇到了nan，并且我的test集人名效果奇差，不知道是哪里的问题 (我并没有改动数据)：
INFO:tensorflow:Finished evaluation at 2019-01-17-15:01:59
INFO:tensorflow:Saving dict for global step 19560: eval_f = nan, eval_precision = nan, eval_recall = nan, global_step = 19560, loss = 30.386103
INFO:tensorflow:Saving 'checkpoint_path' summary for global step 19560: /home/wei/BERT-BiLSTM-CRF-NER/output/model.ckpt-19560
INFO:tensorflow:evaluation_loop marked as finished
INFO:tensorflow:***** Predict results *****
INFO:tensorflow:  eval_f = nan
INFO:tensorflow:  eval_precision = nan
INFO:tensorflow:  eval_recall = nan
INFO:tensorflow:  global_step = 19560
INFO:tensorflow:  loss = 30.386103
INFO:tensorflow:Calling model_fn.
INFO:tensorflow:Running infer on CPU
INFO:tensorflow:*** Features ***
INFO:tensorflow:  name = input_ids, shape = (?, 128)
INFO:tensorflow:  name = input_mask, shape = (?, 128)
INFO:tensorflow:  name = label_ids, shape = (?, 128)
INFO:tensorflow:  name = segment_ids, shape = (?, 128)
shape of input_ids (?, 128)
INFO:tensorflow:**** Trainable Variables ****
INFO:tensorflow:Done calling model_fn.
INFO:tensorflow:Graph was finalized.
INFO:tensorflow:Restoring parameters from /home/wei/BERT-BiLSTM-CRF-NER/output/model.ckpt-19560
INFO:tensorflow:Running local_init_op.
INFO:tensorflow:Done running local_init_op.
processed 214542 tokens with 7450 phrases; found: 197725 phrases; correct: 7061.
accuracy:  10.90%; precision:   3.57%; recall:  94.78%; FB1:   6.88
              LOC: precision:  95.70%; recall:  95.06%; FB1:  95.38  3441
              ORG: precision:  92.09%; recall:  91.37%; FB1:  91.73  2149
              PER: precision:   0.93%; recall:  98.30%; FB1:   1.84  192135
有人遇到过吗？

**Comment by macanv** [comment]
不知道为啥，我使用多种数据运行，都没出现过NON，能给出其他方法的baseline 再来比较么？

**Comment by WaNePr** [comment]
> 我也遇到了类似问题，正在排查原因。
> 我是在多核CPU上运行的，GPU环境不方便升级到CUDA9.0。

请问你最终解决了这个问题么，测试结果NaN

**Comment by WaNePr** [comment]
@makotohomula 
> 我遇到了类似问题，清除output里面的输出，重新运行时得到了解决

请问你清除了哪些output的文件

**Comment by WaNePr** [comment]
> 我也遇到了nan，并且我的test集人名效果奇差，不知道是哪里的问题 (我并没有改动数据)：
> INFO:tensorflow:Finished evaluation at 2019-01-17-15:01:59
> INFO:tensorflow:Saving dict for global step 19560: eval_f = nan, eval_precision = nan, eval_recall = nan, global_step = 19560, loss = 30.386103
> INFO:tensorflow:Saving 'checkpoint_path' summary for global step 19560: /home/wei/BERT-BiLSTM-CRF-NER/output/model.ckpt-19560
> INFO:tensorflow:evaluation_loop marked as finished
> INFO:tensorflow:***** Predict results *****
> INFO:tensorflow: eval_f = nan
> INFO:tensorflow: eval_precision = nan
> INFO:tensorflow: eval_recall = nan
> INFO:tensorflow: global_step = 19560
> INFO:tensorflow: loss = 30.386103
> INFO:tensorflow:Calling model_fn.
> INFO:tensorflow:Running infer on CPU
> INFO:tensorflow:*** Features ***
> INFO:tensorflow: name = input_ids, shape = (?, 128)
> INFO:tensorflow: name = input_mask, shape = (?, 128)
> INFO:tensorflow: name = label_ids, shape = (?, 128)
> INFO:tensorflow: name = segment_ids, shape = (?, 128)
> shape of input_ids (?, 128)
> INFO:tensorflow:**** Trainable Variables ****
> INFO:tensorflow:Done calling model_fn.
> INFO:tensorflow:Graph was finalized.
> INFO:tensorflow:Restoring parameters from /home/wei/BERT-BiLSTM-CRF-NER/output/model.ckpt-19560
> INFO:tensorflow:Running local_init_op.
> INFO:tensorflow:Done running local_init_op.
> processed 214542 tokens with 7450 phrases; found: 197725 phrases; correct: 7061.
> accuracy: 10.90%; precision: 3.57%; recall: 94.78%; FB1: 6.88
> LOC: precision: 95.70%; recall: 95.06%; FB1: 95.38 3441
> ORG: precision: 92.09%; recall: 91.37%; FB1: 91.73 2149
> **PER: precision: 0.93%; recall: 98.30%; FB1: 1.84 192135**
> 有人遇到过吗？

@newwaylw 请问你解决了这个问题么？
